examples/qml_features/rust/src/custom_base_class.rs

```diff
@@ -53,6 +60,12 @@ mod ffi {
         fn end_reset_model(self: Pin<&mut CustomBaseClassQt>);
     }
 
+    // TODO: have a way to generate the type specific code
```

Comment:
this has problems as you hit `only traits defined in the current crate can be implemented for types defined outside of the crate` and the same for primitive types like `i32`.

When trying to for `impl QSetElement for i32` and `unsafe impl ExternType for QSetI32` (`QSet<i32>`)

Structs can be used to create intermediate types and then deref to get back, but doesn't quite work with the CXX bridge yet...

Reply:
Solved, with some templating and trait fun :-)